Emby.Server.Implementations/Networking/NetworkManager.cs

```diff
@@ -636,6 +636,65 @@ public bool TryParseIpAddress(string ipAddress, out IpAddressInfo ipAddressInfo)
             return false;
         }
 
+        public bool IsInSameSubnet(IpAddressInfo address1, IpAddressInfo address2, IpAddressInfo subnetMask)
+        {
+             IPAddress network1 = GetNetworkAddress(ToIPAddress(address1), ToIPAddress(subnetMask));
+             IPAddress network2 = GetNetworkAddress(ToIPAddress(address2), ToIPAddress(subnetMask));
+             return network1.Equals(network2);
+        }
+
+        private IPAddress GetNetworkAddress(IPAddress address, IPAddress subnetMask)
+        {
+            byte[] ipAdressBytes = address.GetAddressBytes();
+            byte[] subnetMaskBytes = subnetMask.GetAddressBytes();
+
+            if (ipAdressBytes.Length != subnetMaskBytes.Length)
+                throw new ArgumentException("Lengths of IP address and subnet mask do not match.");
+
+            byte[] broadcastAddress = new byte[ipAdressBytes.Length];
+            for (int i = 0; i < broadcastAddress.Length; i++)
+            {
+                broadcastAddress[i] = (byte)(ipAdressBytes[i] & (subnetMaskBytes[i]));
+            }
+            return new IPAddress(broadcastAddress);
+        }
+
+        public IpAddressInfo GetLocalIpSubnetMask(IpAddressInfo address)
+        {
+            NetworkInterface[] interfaces;
+            IPAddress ipaddress = ToIPAddress(address);
+
+            try
+            {
+                var validStatuses = new[] { OperationalStatus.Up, OperationalStatus.Unknown };
+
+                interfaces = NetworkInterface.GetAllNetworkInterfaces()
+                    .Where(i => validStatuses.Contains(i.OperationalStatus))
+                    .ToArray();
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Error in GetAllNetworkInterfaces");
+                return null;
+            }
+
+            foreach (NetworkInterface ni in interfaces)
+            {
+                var addr = ni.GetIPProperties().GatewayAddresses.FirstOrDefault();
```

Comment:
seems you don't use `addr` down below, so maybe just
```suggestion
                if (ni.GetIPProperties().GatewayAddresses.FirstOrDefault() != null)
```
?